patches/server/0889-Fix-NBT-pieces-overriding-a-block-entity-during-worl.patch

```diff
@@ -0,0 +1,37 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: etil2jz <blanchot.arthur@protonmail.ch>
+Date: Sat, 2 Apr 2022 23:29:24 +0200
+Subject: [PATCH] Fix NBT pieces overriding a block entity during worldgen
+ deadlock
+
+
+diff --git a/src/main/java/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate.java b/src/main/java/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate.java
+index c5827e4d870a0bba483649d54cae23072eab2b18..332d080ad722a0252d2ae8eb83f7697c1323b4ce 100644
+--- a/src/main/java/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate.java
++++ b/src/main/java/net/minecraft/world/level/levelgen/structure/templatesystem/StructureTemplate.java
+@@ -265,7 +265,11 @@ public class StructureTemplate {
+ 
+                         if (definedstructure_blockinfo.nbt != null) {
+                             tileentity = world.getBlockEntity(blockposition2);
+-                            Clearable.tryClear(tileentity);
++                            // Paper start - Fix NBT pieces overriding a block entity during worldgen deadlock
++                            if (!(world instanceof net.minecraft.world.level.WorldGenLevel)) {
++                                Clearable.tryClear(tileentity);
++                            }
++                            // Paper end
+                             world.setBlock(blockposition2, Blocks.BARRIER.defaultBlockState(), 20);
+                         }
+ 
+@@ -380,7 +384,11 @@ public class StructureTemplate {
+                         if (pair1.getSecond() != null) {
+                             tileentity = world.getBlockEntity(blockposition6);
+                             if (tileentity != null) {
+-                                tileentity.setChanged();
++                                // Paper start - Fix NBT pieces overriding a block entity during worldgen deadlock
++                                if (!(world instanceof net.minecraft.world.level.WorldGenLevel)) {
++                                    tileentity.setChanged();
++                                }
++                                // Paper end
```

Comment:
I assume its a non-issue that setChanged isn't called marking the chunk as unsaved? I don't know the particulars but I'm guessing during worldgen that doesn't matter as its all unsaved anyways.

Reply:
That's what I was thinking of, but yeah it seems to be a non-issue doing this way.
Also, I tried to reproduce with datapack attached on Mojira, no deadlock happened and I didn't encounter any issue that may have been introduced doing this way.